src/libraries/SwapFeeLibrary.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.20;
+
+import {PoolKey} from "../types/PoolKey.sol";
+
+library SwapFeeLibrary {
+    using SwapFeeLibrary for uint24;
+
+    /// @notice Thrown when the static or dynamic fee on a pool is exceeds 100%.
+    error FeeTooLarge();
+
+    uint24 public constant STATIC_FEE_MASK = 0x0FFFFF;
+    uint24 public constant DYNAMIC_FEE_FLAG = 0x800000;
+
+    // the swap fee is represented in hundredths of a bip, so the max is 100%
+    uint24 public constant MAX_SWAP_FEE = 1000000;
+
+    function isDynamicFee(uint24 self) internal pure returns (bool) {
+        return self & DYNAMIC_FEE_FLAG != 0;
```

Comment:
curious on your thoughts around strictly forcing `key.fee = 0x800000` for dynamic-fee pools? would create uniformity for all dynamic fee pools

is there a reason why its `0x800000` and not `0xF00000`?